src/components/PopupWithForm.js

```diff
@@ -6,8 +6,28 @@ export default class PopupWithForm extends Popup {
     this.submit = submitFunction;
     this.form = this.modal.querySelector(".form");
     this.inputs = this.form.querySelectorAll(".form__input");
+    this.button = this.form.querySelector(".form__submit-button");
+    this.buttonText = this.button.textContent;
   }
 
+  renderLoading(isLoading) {
+    if (isLoading) {
+      if (this.form.id === "form-delete-card") {
```

Comment:
Needs correction:

1) The PopupWithForm  should not know anything about the specific popups
2) For handling the confirmation you should create a separate class PopupWithConfirmation

Reply:
Still needs correction